Formula/extism.rb

```diff
@@ -0,0 +1,33 @@
+class Extism < Formula
+  desc "Extend anything with WebAssembly"
+  homepage "https://extism.org"
+  VERSION = "v0.0.1-rc.4".freeze
+  url "https://github.com/extism/extism/archive/refs/tags/#{VERSION}.tar.gz"
+  sha256 "8c60d879d2d92bdadd174889c41a22983f4456d2baf3eb2b67451e71aa5f1ac1"
+  license "BSD-3-Clause"
+
+  depends_on "rust" => :build
+
+  resource "cli" do
+    url "https://raw.githubusercontent.com/extism/cli/main/extism_cli/__init__.py"
+    # TODO: enable sha once we have releases of the cli
+    # sha256 "dde4308c9df551d77e6ff9f3ae116777433a5870bedb6be4a96567970154a79a"
+  end
+
+  def install
+    # it seems i need this make build before make install
+    # otherwise i was getting compile errors on serde
+    system "make", "build"
+    with_env DEST: prefix do
+      system "make", "install"
+    end
+    resource("cli").stage do
+      mkdir_p bin.to_s
+      bin.install "__init__.py" => "extism"
+    end
+  end
+
+  test do
+    system "extism", "info"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.